### PR TITLE
Mp story2

### PR DESCRIPTION
**Pull request by elstevens23**
This story hides the hiring partner attendence section on the Events Page on some events (if, when creating the event, the 'available to hiring partners' checkbox is not checked).

### CODE REVIEW

File: public/js/addEvent.js [hunk 1]
```diff
@@ -47,6 +46,7 @@ let getCompletedFormData = () => {
     let data = {
         name: eventForm.elements['event-name'].value,
         category: eventForm.elements['event-category'].value,
+        availableToHP: eventForm.elements['available-to-HP'].checked ? 1 : 0,
```
Comment (mporam) [inline public/js/addEvent.js:49]: do you need the ternary? it will contain true or false anyway

File: public/js/eventsPage.js [hunk 2]
```diff
@@ -228,28 +228,32 @@ async function eventGenerator(event, hiringPartners) {
...
     }
 
-    eventInformation += `<div class="event-attendees">`
-    eventInformation += `<div class="hiring-partners col-xs-12 col-md-6" data-eventId='${event.id}'></div>`;
-    if (!isPastPage) {
-        eventInformation += `
-            <div class='addHiringPartner col-xs-12 col-md-6'>
-                <h5>Add attendees</h5>
-                <form class='addHiringPartnerForm' id='${event.id}'>
+    if (event.availableToHP == 1) {
```
Comment (mporam) [inline public/js/eventsPage.js:231]: dont need the comparison, 1 is true anyway, so just do `if (event.availableToHP)`

File: templates/addEvent.phtml [hunk 3]
```diff
@@ -24,6 +24,11 @@
                 <?php echo \Portal\ViewHelpers\EventCategoryViewHelper::eventCategoryDropdown($data); ?>
             </select>
 
+            <div class="form-group">
+                <input class="" name="available-to-HP" type="checkbox"/>
```
Comment (mporam) [inline templates/addEvent.phtml:28]: why an empty class?

File: templates/addEvent.phtml [hunk 4]
```diff
@@ -24,6 +24,11 @@
...
+                <label for="available-to-HP">Is this event available to hiring partners?</label>
```
Comment (mporam) [inline templates/addEvent.phtml:29]: `for` must link to an `id`

File: templates/addEvent.phtml (hunk 3 above)
Comment (mporam) [inline templates/addEvent.phtml:28]: also if you put the input INSIDE the label, it makes the label clickable

File: tests/Entities/EventEntityTest.php [hunk 5]
```diff
@@ -153,8 +156,57 @@ public function testNewEventEntitySuccess()
...
+        $result = $name->getAvailableToHP();
+        $this->assertEquals($result, 0);
+    }
+
+    public function testConstructorFailure()
+    {
+        $this->expectException(Exception::class);
+        new EventEntity(
+            1,
+            'hiring eventhhhhfhfgsgshshshshshhshshshshhshshshjakhdkahfksdhflsahfkhaslfkhskfhlkhfklahfdfkhgkaldhfsf
```
Comment (mporam) [inline tests/Entities/EventEntityTest.php:186]: obviously

**Review by mporam — CHANGES_REQUESTED** [review]

File: public/js/addEvent.js (hunk 1 above)
Comment (jennymegan) [reply public/js/addEvent.js:49]: false wasn't passing in as zerothrough mysql - this made it happy

**Review by mporam — APPROVED** [review]

**Pull request merged**